mmit/core/piecewise_function.cpp

```diff
@@ -19,161 +19,155 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #include "piecewise_function.h"
 
+#define TOL 1e-7
```

Comment:
We might be able to avoid it by writing function type-specific logic in functions such as `is_increasing`, `is_decreasing`, etc.